apiserver/src/client/webclient.rs

```diff
@@ -45,20 +45,32 @@ pub trait APIServerClient {
     async fn create_bottlerocket_shadow(
         &self,
         req: CreateBottlerocketShadowRequest,
+        namespace: String,
     ) -> Result<BottlerocketShadow>;
     async fn update_bottlerocket_shadow(
         &self,
         req: UpdateBottlerocketShadowRequest,
+        namespace: String,
```

Comment:
> it's not generically necessary for all implementations

Won't all implementations of `update_bottlerocket_shadow` require the namespace since the `bottlerocketshadow` is a namespaced CRD?

We got away with this before because the controller _only_ knew how to operate on `brs` in the hard coded `brupop-bottlerocket-aws` namespace. If a cluster operator tried to deploy `brs` resources to any other namespace, the entire stack breaks.

In my mind, because the CRD is namespaced, `brs` resources are tightly coupled to the namespace they are deployed in and the API should reflect that as well

Reply:
The way I'm thinking about it is that the namespace here identifies the location of the APIServer itself, so it's more of a client configuration detail. It's contrived for sure, but the test implementation is an example of one that doesn't need the configuration, but would now be required to ignore `namespace` or assert that the value corresponds to an environment-derived value.

Reply:
> In my mind, because the CRD is namespaced, brs resources are tightly coupled to the namespace they are deployed in and the API should reflect that as well

Isn't this also reflected if we make it a member of the client constructor? It's not possible for the API server to modify shadows outside of its namespace, I don't think there's ever a case where you pass a different namespace in this value that succeeds.

Reply:
>  the test implementation is an example of one that doesn't need the configuration, but would now be required to ignore namespace or assert that the value corresponds to an environment-derived value.

That's an excellent point!!! Yes, this can be refactored out to the constructor!

Reply:
Ended up doing abit of a refactor. Now, when the various client configs are created, we pass down the namespace into subsequent constructors that require namespaced resources. It didn't make sense to have to create a new client config with each query to namespaced resources. That would be a costly operation. Instead, we can now access the `namespace` data member where it makes sense.